robot/rover/MobilePlatform/src/APA102.cpp

```diff
@@ -0,0 +1,54 @@
+#include "APA102.h"
+
+#define START_FRAME_SIZE 4
+
+APA102::APA102(int numOfLEDs, int mosiPin, int clockPin)
+                        : numOfLEDs(numOfLEDs), mosiPin(mosiPin), clockPin(clockPin) {
+  int sizeEndFrame = numOfLEDs/16;
+  if (sizeEndFrame * 8 < (numOfLEDs / 2)) {
+    sizeEndFrame++;
+  }
+  bufSize = START_FRAME_SIZE + numOfLEDs * 4 + sizeEndFrame;
+  buf = (byte*)malloc(bufSize);
+  for (int i = 0; i < START_FRAME_SIZE; i++) {
+    buf[i] = 0;
+  }
+  for (int i = START_FRAME_SIZE; i < 4 * numOfLEDs + START_FRAME_SIZE; i+= 4) {
+    buf[i + 0] = 0xE0;
+    buf[i + 1] = 0;
+    buf[i + 2] = 0;
+    buf[i + 3] = 0;
+  }
+  for (int i = START_FRAME_SIZE + numOfLEDs * 4; i < bufSize; i++) {
+    buf[i] = 255;
+  }  
+}
+void APA102::send() {
+  //spi->beginTransaction(settingsA);
+  for (int i = 0; i < bufSize; i++) {
+    shiftOut(mosiPin, clockPin, MSBFIRST, buf[i]);
+    //spi->transfer(buf[i]);
```

Comment:
These commented out lines should probably be removed

Reply:
just committed this change